code/game/objects/effects/spawners/random.dm

```diff
@@ -1,531 +1,886 @@
+/**
+ * Base class for all random spawners.
+ */
 /obj/effect/spawner/random
-	name = "Random Object"
-	desc = "This item type is used to spawn random objects at round-start"
+	icon = 'icons/effects/landmarks_static.dmi'
+	icon_state = "loot"
+	layer = OBJ_LAYER
+	/// Stops persistent lootdrop spawns from being shoved into lockers
+	anchored = TRUE
+	/// A list of possible items to spawn e.g. list(/obj/item, /obj/structure, /obj/effect)
+	var/list/loot
+	/// The subtypes AND type to combine with the loot list
+	var/loot_type_path
+	/// The subtypes (this excludes the provided path) to combine with the loot list
+	var/loot_subtype_path
+	/// How many items will be spawned
+	var/spawn_loot_count = 1
+	/// If the same item can be spawned twice
+	var/spawn_loot_double = TRUE
+	/// Whether the items should be distributed to offsets 0,1,-1,2,-2,3,-3.. This overrides pixel_x/y on the spawner itself
+	var/spawn_loot_split = FALSE
+	/// Whether the spawner should spawn all the loot in the list
+	var/spawn_all_loot = FALSE
+	/// The chance for the spawner to create loot (ignores spawn_loot_count)
+	var/spawn_loot_chance = 100
+	/// Determines how big of a range (in tiles) we should scatter things in.
+	var/spawn_scatter_radius = 0
+	/// Whether the items should have a random pixel_x/y offset (maxium offset distance is ±16 pixels for x/y)
+	var/spawn_random_offset = FALSE
+	///does the spawned structure need to retain its direction on spawn?
+	var/spawn_with_original_direction = FALSE
+
+/obj/effect/spawner/random/Initialize(mapload)
+	. = ..()
+	spawn_loot()
+	qdel(src)
+
+///If the spawner has any loot defined, randomly picks some and spawns it. Does not cleanup the spawner.
+/obj/effect/spawner/random/proc/spawn_loot(lootcount_override)
+	if(!prob(spawn_loot_chance))
+		return
+
+	var/list/spawn_locations = get_spawn_locations(spawn_scatter_radius)
+	var/spawn_loot_count = isnull(lootcount_override) ? src.spawn_loot_count : lootcount_override
+
+	if(spawn_all_loot)
+		spawn_loot_count = INFINITY
+		spawn_loot_double = FALSE
+
+	if(loot_type_path)
+		loot += typesof(loot_type_path)
+
+	if(loot_subtype_path)
+		loot += subtypesof(loot_subtype_path)
+
+	if(loot?.len)
```

Comment:
I'm not sure what you mean here, if I said 
	if(!loot_subtype_path)
		return
it would stop spawning if any item didn't have a subtype path, and most spawners don't.